docs/proposals/multiple-sources-for-applications-ui.md

```diff
@@ -0,0 +1,226 @@
+This can be treated as a separate, independent proposal.
+
+Other related changes pertain to the Last Synced Details. The Sync Details panel needs to be updated to show sync info
+from multiple sources.  See [Issue 13215](https://github.com/argoproj/argo-cd/issues/13215).
+
+### New App Dialog
+
+_Current Behavior:_ 
+
+The dialog currently allows users to ‘quickly’ create a single source application..
+
+_Proposed Changes:_
+
+Make the form view of the dialog support adding, updating and viewing of multiple sources. The issue with the current 
+single source New App wizard is that it can lead to loss of “input” provided by the user. The content in the form-based 
+editor and the YAML editor (accessed via the Edit as YAML button) must match. If the user provides multiple sources in 
+the YAML editor, and then switches back to the form view, the form will only show the first source. The other sources 
+are effectively ‘lost’. Furthermore, if the user switches back to the YAML editor, only one source will be shown as well.
+
+The design and changes (React components) from the new Sources tab can likely be reused in this dialog.
+
+Other Changes. This includes the underlying plumbing to create an app using the Sources field of the Application CR, so that the 
+deprecated Source field can be removed in the future.
+
+
+
+### Use cases
+
+The use cases involves those areas in the UI where the current source is displayed. These have been described
+in the Summary and Proposal sections.
+
+
+### Implementation Details
+
+The implementation plan can be divided into different stages. Read-only capability can be provided first and it will 
+be the safest change. The UI currently is not showing all the sources for the multi-source application so this should 
+be the highest priority. (Before you can edit, you have to first display it.)
+
+Here are the general enhancements to be implemented (Upstream issues to be opened if not already):
+
+1. Create new Sources tab to replace Parameters tab so that all sources can be displayed (Read-only)
+2. Update History and Rollback to show a summary of all sources of an application
+   As mentioned above, this is already covered by Jorge’s [PR](https://github.com/argoproj/argo-cd/pull/14124)
+3. Add _Add Source_ and _Delete Source_ buttons to Sources tab. This will depend on #1 above. (Update and Delete)
+4. Update New App dialog. (Creation)
+   - Support adding multiple sources in New App dialog. (This will likely depend on the Components from #1 and #3)
+   - Use Sources field instead of Source field. Clean up code.
+
+### Security Considerations
+None
+
+### Risks and Mitigations
+None
+
+### Upgrade / Downgrade Strategy
+If downgraded, the UI will revert to showing just the first source.
+
+## Drawbacks
+None
+
+## Open Questions
+
+Supporting multiple sources in the New App dialog may not be ‘worth’ the effort? The drawback is that switching from the 
+YAML editor and form editor can lead to loss of information.
```

Comment:
I think it is worth having an issue for tracking and if there are users of Argo where the creation of multi sources through the UI is critical for them, they can contribute and implement it.

Reply:
Ok, I'll open an issue for any outstanding enhancement work that could be done in the future.